xarray/tests/test_cftimeindex.py

```diff
@@ -988,6 +988,22 @@ def test_cftimeindex_calendar_property(calendar, expected):
     assert index.calendar == expected
 
 
+@requires_cftime
+def test_cftime_noleap_with_str():
+    # https://github.com/pydata/xarray/issues/9138
+    time = pd.date_range("2000-01-01", "2006-01-01", freq="D")
+    temperature = np.ones(len(time))
+    da = xr.DataArray(
+        data=temperature,
+        dims=["time"],
+        coords=dict(
+            time=time,
+        ),
+    )
+    da_noleap = da.convert_calendar("noleap")
+    da_noleap.sel(time=slice("2001", "2002"))
```

Comment:
I'm not really sure what should be asserted here. I mostly just "simplified" the test case to the bare essential.

Reply:
Thanks!  I think the right thing to assert is that the index is a `CFTimeIndex` or `DatetimeIndex` (depending on the context).  That should ensure that partial string based indexing and other time indexing features are retained on the output—I updated this in 91e8b0d.